Scicd 580

larry-tordsen opened the pull request:
## Summary and Scope

A number of minor fixes related to SCICD-580

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves https://jira-pro.its.hpecorp.net:8443/browse/SCICD-580

### Tested on:
Frigg

ltordsen committed:
SCICD-580 Minor Fixes

Change some of the help output.

Also, print the aborted activity when running the abort command.

ltordsen committed:
SCICD-580: Minor Fixes

Pass the comment in the payload for the abort messages.

ltordsen committed:
SCICD-580: bootprep flags

Allow the user to enter a normal path for the bootprep parameters.  Copy
the files (or directories) into the media directory.

larry-tordsen commented on lib/Activity.py:
```diff
@@ -526,14 +528,18 @@ def run_stages(self, config):
         if bp_config_managed:
             payload["input_parameters"]["bootprep_config_managed"] = bp_config_managed
 
+        bp_config_dir = config.args.get("relative_bootprep_config_dir", None)
```
Not sure if this should be passed to the backend.  The backend isn't using it.  This might just be a relic of the cne-installer.

joshua-plautz reviewed (approved):
Someone will need to make a mod to docs-csm to adjust the cli docs with the new command line arg

deantroe commented:
> Someone will need to make a mod to docs-csm to adjust the cli docs with the new command line arg

On it.

deantroe commented on iuf:
```diff
@@ -427,13 +437,15 @@ def main():
-        help="""Override list used to target specific role_subrole(s) only when rolling out management nodes.""")
+        help="""Override list used to target specific role_subrole(s) only
+        when rolling out management nodes.  Arguments should be xnames or HSM
+        node groups.""")
```
This was fine before: management rollout uses role_subrole(s) only.

deantroe commented:
This should have been addressed in a previous PR, but since this one is open...

I think we should change the help output for abort from this:
```Abort a previously paused or failed IUF session for a given activity.```
...to this:
```Abort an IUF session for a given activity.```

Also, we should explain what ```comment``` is for in the ```iuf abort -h``` output.

deantroe reviewed (changes requested):
Made a few comments.

larry-tordsen commented:
> This should have been addressed in a previous PR, but since this one is open...
> 
> I think we should change the help output for abort from this: `Abort a previously paused or failed IUF session for a given activity.` ...to this: `Abort an IUF session for a given activity.`
> 
> Also, we should explain what `comment` is for in the `iuf abort -h` output.

I'm going to hide the `--comment` flag for now.  It's not working right on the backend.

ltordsen committed:
SCICD-580: Incorporate review comments

Incorporate review comments.  Also add a force flag to force the termination
of an activity (the default behavior is to wait for the current running
stage to finish.
